### PR TITLE
define Automaton interface locally

### PR DESCRIPTION
**Pull request by mschoch**
In Go we prefer to define interfaces where we consume them.
Since the Automaton is now a core part of how scorch allows
one to iterate a term dictionary, this definition belongs here.

As a side-effect, this removes dependency on the vellum package
from the API (although both Bleve and Zap implementation code
continues to use the vellum package).

### CODE REVIEW

**Comment by mschoch** [comment]
As this is a breaking change, once merged we will tag `v2.0.0`

File: go.mod [hunk 1]
```diff
@@ -5,6 +5,6 @@ go 1.13
 require (
 	github.com/RoaringBitmap/roaring v0.4.23
 	github.com/blevesearch/bleve_index_api v1.0.0
-	github.com/couchbase/vellum v1.0.2
+	github.com/davecgh/go-spew v1.1.1 // indirect
```
Comment (abhinavdangeti) [inline go.mod:8]: Interesting that this should add an indirect dependency.

**Review by abhinavdangeti — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (mschoch) [reply go.mod:8]: Wow, go modules continues to confuse.  I mean we're always running go mod tidy, and nothing else has changed... So WTF?

Comment (mschoch) [reply go.mod:8]: Actually I have a theory...

Perhaps vellum and roaring both use go-spew, but vellum had it as a part of it's go.mod.  Now by removing vellum, there is no reference for it, so it becomes an indirect reference for our project???

Comment (mschoch) [reply go.mod:8]: Oh wait, except we wrote vellum and have never heard of go-spew, lol

Comment (mschoch) [reply go.mod:8]: Maybe something was cached weird for couchbase/vellum locally, so it never added it.  But when we switched to blevesearch/vellum it realized it needed it?

Comment (abhinavdangeti) [reply go.mod:8]: That'd explain it, but vellum doesn't have a go-spew in it's list of direct dependencies.

Comment (abhinavdangeti) [reply go.mod:8]: So looks like one of vellum's dependencies references it, it's part of vellum's go.sum. Your theory holds.

**Pull request merged**